strings/base_coroutine_foundation.h

```diff
@@ -512,6 +549,12 @@ namespace winrt::impl
         {
             slim_lock_guard const guard(m_lock);
             WINRT_ASSERT(m_status.load(std::memory_order_relaxed) == AsyncStatus::Started || m_status.load(std::memory_order_relaxed) == AsyncStatus::Canceled);
+
+            if (m_is_noexcept)
+            {
+                winrt::terminate();
```

Comment:
maybe a comment here to explain how to diagnose the problem

Reply:
I think the new code makes it self-evident why we are terminating:

```cpp
if (m_terminate_on_unhandled_exception)
{
    winrt::terminate();
}
```